pkgs/development/compilers/hip/default.nix

```diff
@@ -0,0 +1,173 @@
+{ stdenv
+, binutils-unwrapped
+, clang
+, clang-unwrapped
+, cmake
+, compiler-rt
+, fetchFromGitHub
+, fetchpatch
+, file
+, lib
+, lld
+, llvm
+, makeWrapper
+, perl
+, python
+, rocclr
+, rocm-comgr
+, rocm-device-libs
+, rocm-opencl-runtime
+, rocm-runtime
+, rocm-thunk
+, rocminfo
+, writeText
+}:
+
+stdenv.mkDerivation rec {
+  name = "hip";
+  version = "4.3.1";
+  src = fetchFromGitHub {
+    owner = "ROCm-Developer-Tools";
+    repo = "HIP";
+    rev = "rocm-${version}";
+    sha256 = "sha256-dUdP32H0u6kVItS+VUE549vvxkV1mSN84HvyfeK2hEE=";
+  };
+
+  # FIXME: https://github.com/ROCm-Developer-Tools/HIP/issues/2317
+  postPatch = ''
+    cp ${rocm-opencl-runtime.src}/amdocl/cl_vk_amd.hpp amdocl/
```

Comment:
I don't have this fix in the nixos-rocm overlay, but 4.3.0 is building fine. Any idea?

Reply:
Well, this is 4.3.1 so perhaps the issue happened on the 0 -> 1 patch changes?

Reply:
Interesting, even for 4.3.0 the Gentoo package needs this: https://github.com/gentoo/gentoo/blob/ab005b2b0219f266a90891fa3cce8253014927db/dev-util/hip/hip-4.3.0.ebuild#L69

Reply:
@acowley Am I cleared to merge this even with this fix?

Reply:
Sure!